src/wet-manager.ts

```diff
@@ -0,0 +1,130 @@
+        //URL address of file on server
+        let url = WetManager.basePath + filename;
+        let parsedUrl = WetManager.url.parse(url);
+
+        let dirArray = WetManager.path.dirname(parsedUrl.path).split('/');
+        //Segment, aka subfolder in our wet directory
+        let segment;
+        if (dirArray.length > 2) {
+            segment = dirArray[dirArray.length - 2];
+        } else {
+            segment = 'unknown';
+        }
+
+        //filename in folder
+        let localFilename = WetManager.path.basename(parsedUrl.path);
+        //complete file path on drive
+        let filepath = WetManager.path.join(
+            WetManager.path.normalize(WetManager.wetFolder),
+            segment,
+            localFilename
+        );
+
+        console.log("File path: " + filepath);
+
+        WetManager.fs.exists(filepath, function(exists) {
+            if(exists) {
+                //Read existing file from file system as decompressed stream for callback
+                console.log('exists, opening from fs');
+                let fileReadStream = WetManager.fs.createReadStream(filepath);
+                callback(null, Unpacker.decompressGZipStream(fileReadStream));
+
+            } else {
+                let dirPath = WetManager.path.join(
+                    WetManager.path.normalize(WetManager.wetFolder),
+                    segment
+                );
+
+                //Create segment directory if not exists
+                if (!WetManager.fs.existsSync(dirPath)){
+                    WetManager.fs.mkdirSync(dirPath);
+                    console.log('Created directory ' + dirPath);
+                }
+
+                //Download file, store compressed on disk and execute callback with decompressed stream
+                console.log('doesnt exist, starting download');
+                WetManager.getResponse(url, function(err, resp) {
+                    if(err) {
+                        callback(err);
+                        return;
+                    }
+                    let outputFile = WetManager.fs.createWriteStream(filepath);
+                    resp.pipe(outputFile);
+                    let decompressed = Unpacker.decompressGZipStream(resp);
+                    decompressed.on('error', err => {
+                        console.log(err);
+                        outputFile.close();
+                        WetManager.fs.unlinkSync(filepath);
+                    }).on('end', () => {
+                        console.log('Download finished');
+                        outputFile.close();
+                    });
+
+
+                    callback(null, decompressed);
+                });
+
+            }
+        });
+    };
+
+    /**
+     * Download file via https or http
+     * @param fileURL       URL of the file
+     * @param callback      Callback function taking IncomingMessage stream as a parameter
+     */
+    private static getResponse(fileURL : string,
```

Comment:
I would actually suggest we cut out Downloader entirely which is why I reused that function. I think with wetmanager working we won't need the Downloader module anymore. I created it as a new module because it offers a different functionality than Downloader, basically an additional layer of abstraction.